src/auth/auth.service.ts

```diff
@@ -29,12 +29,20 @@ export class AuthService {
     private readonly oidcHelperService: OidcHelperService,
     private readonly bypassService: BypassService,
     private configService: ConfigService,
-  ) {}
+  ) { }
 
   async determinePolicy(userId: string): Promise<PolicyResponse> {
     this.logger.debug('Starting determinePolicy', { userId });
 
     userId = userId ? userId.toUpperCase() : userId;
+
+    const appStatus = this.configService.get('app.appStatus');
+    const maintenanceBypassUsers = this.configService.get('app.maintenanceBypassUsers');
+
+    if (appStatus === "TEST" && (!maintenanceBypassUsers.includes(userId) && !maintenanceBypassUsers.includes(userId?.toLowerCase()))) {
+      throw new EaseyException(new Error('The server is temporarily unable to service your request due to maintenance. Please try again later.'), HttpStatus.SERVICE_UNAVAILABLE);
+    }
```

Comment:
You could use the `validateUserMaintenanceAblity` method here:
```suggestion
    this.validateUserMaintenanceAblity({ userId } as UserDTO);
```